lib/mayaUsd/utils/editRouterContext.h

```diff
@@ -77,6 +83,13 @@ class MAYAUSD_CORE_PUBLIC OperationEditRouterContext : public StackedEditRouterC
      */
     OperationEditRouterContext(const PXR_NS::TfToken& operationName, const PXR_NS::UsdPrim& prim);
 
+    /*! \brief Route to the given stage and layer.
+     *  Should be used in undo to ensure the same target is used as in the initial execution.
+     */
+    OperationEditRouterContext(
+        const PXR_NS::UsdStagePtr&    stage,
+        const PXR_NS::SdfLayerHandle& layer);
```

Comment:
These new constructors allow creating a router context directly with the desired layer instead of calling the edit router. As explained in the comment above, this is to support undo and redo correctly, which both want to use the same target as was decided when executed the first time. We don't want to make different decisions on undo and redo.